### DISCUSSION TITLE
Open SourceMRI Projects relations

### DISCUSSION THREAD

**Discussion by M4a1x** [vnegnev/marcos_extras#11] [General]
Hi everyone!

I'm new to the scene and was wondering what the relations between the different open source MRI projects that I found are. Specifically this project "MaRCoS", the [open source imaging initiative](https://github.com/opensourceimaging) and the [ocra mri](https://openmri.github.io/ocra/) (and maybe its relation to `flocra`. Maybe also how `pulseq` is used?

Is there a single currently recommended project? Is one superseding the other? Or are they targeting different things? I'd be happy about a quick overview.

For context: I'm trying to build a low-cost open-source tabletop NMR spectrometer and evaluating which software I should use. The RedPitaya/MaRCoS combination seems very appealing!

**Comment by vnegnev** [comment]
Hi Max, sorry about the delay (I wasn't getting notifications for some reason).

Quick history: MaRCoS was originally a fork of OCRA mainly to support the SDRLab-122. As part of this we rewrote the server and Python client, then added support for the GPA-FHDO, then replaced the rest of the OCRA FPGA firmware entirely. Now OCRA is entirely separate to MaRCoS and they're developing along different lines. I don't personally know much about the open source imaging initiative, but it looks like they're mostly hardware-focused based on the GitHub repo - they are probably broadly compatible with the Red Pitaya but I haven't investigated. `flocra` is the old name for the MaRCoS FPGA firmware, standing for 'flow-based OCRA', but this was confusing since there's no shared code and the protocol is different so we've renamed `flocra` to `marga` since (the old name is still floating around though! If you see it in code, let me know and I'll update it).

Broadly I'd say MaRCoS owes a huge amount to OCRA and the people behind it, but their feature set has since diverged from ours so you should take a look at both and see which one fits your needs better (I'm also one of the main MaRCoS developers, so perhaps you should take anything I say with a grain of salt ;) ) Hope that helps!
